service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/AwaitCloneControlledAzureDatabaseResourceFlightStep.java

```diff
@@ -0,0 +1,102 @@
+package bio.terra.workspace.service.resource.controlled.flight.clone.workspace;
+
+import static bio.terra.workspace.common.utils.FlightUtils.*;
+
+import bio.terra.stairway.*;
+import bio.terra.stairway.exception.DatabaseOperationException;
+import bio.terra.stairway.exception.FlightWaitTimedOutException;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.utils.FlightUtils;
+import bio.terra.workspace.service.job.JobMapKeys;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.database.ControlledAzureDatabaseResource;
+import bio.terra.workspace.service.resource.controlled.flight.clone.azure.common.ClonedAzureResource;
+import bio.terra.workspace.service.resource.controlled.model.ControlledResource;
+import bio.terra.workspace.service.resource.model.StewardshipType;
+import bio.terra.workspace.service.resource.model.WsmResourceType;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.model.WsmCloneResourceResult;
+import bio.terra.workspace.service.workspace.model.WsmResourceCloneDetails;
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Wait for the clone controlled database flight to complete and add the result to the appropriate
+ * map
+ */
+public class AwaitCloneControlledAzureDatabaseResourceFlightStep implements Step {
+
+  private final ControlledAzureDatabaseResource resource;
+  private final String subflightId;
+
+  public AwaitCloneControlledAzureDatabaseResourceFlightStep(
+      ControlledAzureDatabaseResource resource, String subflightId) {
+    this.resource = resource;
+    this.subflightId = subflightId;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
```

Comment:
I don't see any "wait" logic here, where does that happen? Am I missing something?

TOL: Could we possibly split out a "handle results" method to make it clearer which parts of this are "waiting for it to end" and which parts are finding and handling the outcomes?

Reply:
I think the AwaitCloneX class has a polling mechanism set up using `context.getStairway().waitForFlight(...)` method which keeps polling for X seconds and then extract results after the polling finishes or when the flight succeeds. Based on the other classes and the implementation I am not really sure how to split it up 🤔